swarm/storage/mru/lookup.go

```diff
@@ -151,27 +150,22 @@ func (u *UpdateLookup) binaryGet(serializedData []byte) error {
 	}
 	cursor += viewLength
 
-	u.Period = binary.LittleEndian.Uint32(serializedData[cursor : cursor+4])
-	cursor += 4
-
-	u.Version = binary.LittleEndian.Uint32(serializedData[cursor : cursor+4])
-	cursor += 4
+	if err := u.Epoch.UnmarshalBinary(serializedData[cursor : cursor+lookup.EpochLength]); err != nil {
+		return err
+	}
+	cursor += lookup.EpochLength
 
 	return nil
 }
 
 // FromValues deserializes this instance from a string key-value store
 // useful to parse query strings
-func (u *UpdateLookup) FromValues(values Values, parseView bool) error {
-	version, _ := strconv.ParseUint(values.Get("version"), 10, 32)
-	period, _ := strconv.ParseUint(values.Get("period"), 10, 32)
+func (u *UpdateLookup) FromValues(values Values) error {
```

Comment:
now why is this not following the binary serialisation pattern you introduced?

Reply:
This is a serializer that saves this struct to a string key/value store. It is used by the server package to compose query strings.